cumulusci/core/config/OrgConfig.py

```diff
@@ -178,3 +180,27 @@ def get_community_info(self, community_name, force_refresh=False):
             )
 
         return self._community_info_cache[community_name]
+
+    @property
+    def installed_packages(self):
+        if not self._installed_packages:
+            response = self.salesforce_client.restful(
+                "tooling/query/?q=SELECT SubscriberPackage.NamespacePrefix, SubscriberPackageVersion.MajorVersion, "
+                "SubscriberPackageVersion.MinorVersion, SubscriberPackageVersion.PatchVersion "
+                "FROM InstalledSubscriberPackage"
+            )
+
+            self._installed_packages = {}
+            for package in response["records"]:
+                sp = package["SubscriberPackage"]
+                spv = package["SubscriberPackageVersion"]
+                # PatchVersion is a 0 on a non-patch version.
+                version = (
+                    f"{spv['MajorVersion']}.{spv['MinorVersion']}.{spv['PatchVersion']}"
+                )
+                self._installed_packages[sp["NamespacePrefix"]] = StrictVersion(version)
```

Comment:
Hmm -- In the 2GP world there could be multiple SPVs sharing the same prefix (or no prefix)

Reply:
That's a really good point and I'm not sure off the cuff what the right strategy is to handle it.

We could use `AllPackageId` alongside the namespace: populate the `AllPackageId` for every package, with or without namespace, and log a warning and do not populate a version for any package where multiple packages are sharing a namespace.

Reply:
I was thinking it might make sense to provide a namespace mapping-based API that intentionally only includes 1GP package versions, and then a separate one that requires using package ids. But I'm not sure we have a way to filter the SPV query to get only 1gp results. Possibly using the Package2ContainerOptions field?

Reply:
I like that idea and it works nicely with your other suggestion to provide a function on `org_config` to abstract away the comparison. I'll investigate to confirm if we can tell 1GPs and 2GPs apart.